modP2pImpl/src/org/aion/p2p/impl1/tasks/TaskStatus.java

```diff
@@ -21,21 +26,28 @@ public TaskStatus(
         this.selfShortId = _selfShortId;
         this.sendMsgQue = _sendMsgQue;
         this.receiveMsgQue = _receiveMsgQue;
+        this.start = _start;
     }
 
     @Override
     public void run() {
-        Thread.currentThread().setName("p2p-ts");
-        String status = this.nodeMgr.dumpNodeInfo(this.selfShortId, p2pLOG.isDebugEnabled());
-
-        if (p2pLOG.isDebugEnabled()) {
-            p2pLOG.debug(status);
-            p2pLOG.debug(
-                    "recv queue[{}] send queue[{}]",
-                    this.receiveMsgQue.size(),
-                    this.sendMsgQue.size());
-        } else if (p2pLOG.isInfoEnabled()) {
-            p2pLOG.info(status);
+        while (start.get()) {
+            try {
+                Thread.sleep(PERIOD_STATUS);
+                String status = this.nodeMgr.dumpNodeInfo(this.selfShortId, p2pLOG.isDebugEnabled());
+
+                if (p2pLOG.isDebugEnabled()) {
+                    p2pLOG.debug(status);
+                    p2pLOG.debug(
+                        "recv queue[{}] send queue[{}]",
+                        this.receiveMsgQue.size(),
+                        this.sendMsgQue.size());
+                } else if (p2pLOG.isInfoEnabled()) {
+                    p2pLOG.info(status);
+                }
+            } catch (Exception e) {
+                p2pLOG.warn("P2p taskStatus exception! ", e);
+            }
```

Comment:
please also add a message when exiting the loop to know that the thread shut down

Reply:
added!